src/silent-blocks/silent-blocks.service.ts

```diff
@@ -0,0 +1,180 @@
+        });
+
+        const transactionsCountBuffer = this.encodeVarInt(
+            transactionBuffers.length,
+        );
+
+        const silentBlockBuffer = Buffer.concat([
+            Buffer.from([silentBlockType]),
+            transactionsCountBuffer,
+            ...transactionBuffers,
+        ]);
+
+        return silentBlockBuffer;
+    }
+
+    /**
+     * Decodes a silent block buffer into an array of transactions.
+     * @param buffer - The buffer to decode.
+     * @returns An array of decoded transactions.
+     */
+    private decodeSilentBlock(buffer: Buffer): Transaction[] {
+        const transactions: Transaction[] = [];
+        let offset = 0;
+
+        // Read the silent block type
+        const silentBlockType = buffer.readUInt8(offset);
+        offset += 1;
+
+        // Read the number of transactions
+        const transactionsCount = this.decodeVarInt(buffer, offset);
+        offset += 1;
+
+        // Process each transaction
+        for (let i = 0; i < transactionsCount; i++) {
+            // Read the txid
+            const txid = buffer.subarray(offset, offset + 32).toString('hex');
+            offset += 32;
+
+            // Read the number of outputs
+            const outputsCount = this.decodeVarInt(buffer, offset);
+            offset += 1;
+
+            // Read the outputs
+            const outputs = [];
+            for (let j = 0; j < outputsCount; j++) {
+                const value = buffer.readBigUInt64BE(offset); // 8-byte value
+                offset += 8;
+
+                const pubKey = buffer
+                    .subarray(offset, offset + 32)
+                    .toString('hex'); // 32-byte pubkey
+                offset += 32;
+
+                const vout = buffer.readUInt32BE(offset); // 4-byte vout
+                offset += 4;
+
+                outputs.push({ value: Number(value), pubKey, vout });
+            }
+
+            // Read the scanTweak
+            const scanTweak = buffer
+                .subarray(offset, offset + 33)
+                .toString('hex'); // 33-byte scan tweak
+            offset += 33;
+
+            // Create the transaction object
+            const transaction: Transaction = {
+                id: txid,
+                outputs,
+                scanTweak,
+
+                // These are not included in the block design, so just use dummy values
+                blockHeight: 0,
+                blockHash: '',
+                isSpent: false,
+            };
+
+            transactions.push(transaction);
+        }
+
+        return transactions;
+    }
+
+    /**
+     * Encodes a variable integer (varInt) into a Buffer.
+     * @param value - The integer value to encode.
+     * @returns A Buffer containing the encoded varInt.
+     */
+    private encodeVarInt(value: number): Buffer {
+        const buffer = Buffer.alloc(1);
+        buffer.writeUInt8(value);
+        return buffer;
+    }
```

Comment:
this is not the right way to encode varint. Please follow standard bitcoin varint encoding and decoding